dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/AbstractJdbcTableManager.java

```diff
@@ -637,30 +615,86 @@ protected void invokeTimeAndLog( String sql, String logMessage )
+    {
+        String mainTableName = mainTable.getTableName();
+        String tempTableName = mainTable.getTempTableName();
+
+        final String[] sqlSteps = {
+            " drop table if exists " + mainTableName + " cascade",
+            " alter table " + tempTableName + " rename to " + mainTableName
+        };
+        final String sql = String.join( ";", sqlSteps ) + ";";
+        log.debug( sql );
         executeSilently( sql );
     }
 
-    /**
-     * Updates table inheritance of a table partition from the temp master table
-     * to the real master table.
-     *
-     * @param partitionTableName the partition table name.
-     * @param tempMasterTableName the temporary master table name.
-     * @param realMasterTableName the real master table name.
-     */
-    private void swapInheritance( String partitionTableName, String tempMasterTableName, String realMasterTableName )
+    private void createAnalyticsTable( AnalyticsTable table )
     {
-        final String sql = "alter table " + partitionTableName + " inherit " + realMasterTableName + ";" +
-            "alter table " + partitionTableName + " no inherit " + tempMasterTableName + ";";
+        createAnalyticsTable( table, null );
+    }
 
-        executeSilently( sql );
+    private void createAnalyticsTable( AnalyticsTable table, AnalyticsTablePartition partition )
+    {
+        tryCreateTableAsPartitionOf( table, partition );
+
+        String tableName = partition == null ? table.getTableName() : partition.getTempTableName();
+        String sqlCreate = "create table if not exists " + tableName + " (";
+        for ( AnalyticsTableColumn col : ListUtils.union( table.getDimensionColumns(), table.getValueColumns() ) )
+        {
+            String notNull = col.getNotNull().isNotNull() ? " not null" : "";
+
+            sqlCreate = sqlCreate + (col.getName() + " " + col.getDataType().getValue() + notNull + ",");
+        }
+
+        sqlCreate = TextUtils.removeLastComma( sqlCreate ) + ")";
+
+        if ( partition == null && getPartitionColumn() != null )
+        {
+            String partitionColumn = getPartitionColumn();
+            sqlCreate += " partition by list(\"" + partitionColumn + "\")";
+        }
+
+        log.debug( String.format( "Creating table: %s, columns: %d", tableName, table.getDimensionColumns().size() ) );
+
+        log.debug( "Created SQL: " + sqlCreate );
+
+        jdbcTemplate.execute( sqlCreate );
+    }
+
+    private void tryCreateTableAsPartitionOf( AnalyticsTable table, AnalyticsTablePartition partition )
+    {
+        if ( partition != null )
+        {
+            String createTableAsPartitionOfSql = "create table if not exists " + partition.getTableName()
+                + " partition of " +
+                table.getTableName() + " for values in " + "(" + partition.getYear() + ")";
+
+            log.debug( String.format( "Creating table: %s, columns: %d", partition.getTableName(),
```

Comment:
With Slf4j the built-in log variable is `{}`, that way you don't have to use String.format.